lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -922,8 +922,43 @@ WitnessChecker::lookupValueWitnessesViaImplementsAttr(
   }
 }
 
-SmallVector<ValueDecl *, 4> 
-WitnessChecker::lookupValueWitnesses(ValueDecl *req, bool *ignoringNames) {
+bool WitnessChecker::isMemberOperator(FuncDecl *decl, Type type) {
+  auto *DC = decl->getDeclContext();
+  auto selfNominal = DC->getSelfNominalTypeDecl();
+
+  // Check the parameters for a reference to 'Self'.
+  bool isProtocol =
+      selfNominal && isa<ProtocolDecl>(selfNominal) &&
+      TC.conformsToProtocol(type, dyn_cast<ProtocolDecl>(selfNominal), DC,
```

Comment:
You know it conforms to the protocol, because you found the operator via name lookup on the type